src/fitPull.ts

```diff
@@ -32,10 +50,13 @@ export class FitPull implements IFitPull {
 			return {status: "localCopyUpToDate", remoteUpdate: null};
 		}
 		if (!localChanges) {
-			localChanges = await this.fit.getLocalChanges();
+			// Scan vault and compare against latest known state
+			localChanges = (await this.fit.getLocalChanges()).changes;
 		}
-		const remoteTreeSha = await this.fit.getRemoteTreeSha(remoteCommitSha);
-		const remoteChanges = await this.fit.getRemoteChanges(remoteTreeSha);
+
+		// Use the remoteCommitSha we already fetched to avoid duplicate API call
+		const { changes: remoteChanges, state: remoteState } = await this.fit.getRemoteChanges(remoteCommitSha);
+
```

Comment:
Oops, I regressed some error handling here so it's not hitting some OctokitHttpError special cases in FitSync anymore... @kody why didn't you warn me?? 😜

I'll send another PR to fix that next.